src/ogma/features/nodeGrouping.ts

```diff
@@ -121,9 +129,55 @@ export class NodeGroupingTransformation {
         }
       },
       nodeSelector: (node) => {
-        // TODO: Tools.isDefined(node.getSubNodes()) is a work around for an ogma issue visible when using image export plugin with Node grouping
-        // remove when updating to Ogma v5.1.x
-        return node.isVirtual() && Tools.isDefined(node.getSubNodes());
+        return node.isVirtual() && !OgmaTools.isGroupCollapsed(node);
+      },
+      // the style will be updated when data object is updated
+      nodeDependencies: {self: {data: true}}
+    });
+
+    this._nodeGroupingCollapsedStyleRule = this._ogma.styles.addRule({
+      nodeAttributes: {
+        text: {
+          content: (node: Node<LkNodeData> | undefined): string | undefined => {
+            return this._getNodeGroupingCaption(node);
+          },
+          style: 'bold'
+        },
+        halo: {
+          width: 20,
+          color: '#e4ebea',
+          strokeColor: '#ccc'
+        },
+        badges: {
+          bottomLeft: (node) => {
+            const numberOfSubNodes = node.getSubNodes()?.size;
+            if (!Tools.isDefined(numberOfSubNodes)) {
+              return;
```

Comment:
is this a use case?

Reply:
normally no as we are specifying that the rule should be applied only to virtual nodes, but you never know with Ogma. 
I will remove it and use null assertion on getSubNodes